test/unit-test/fam-api/fam_progress_mt.cpp

```diff
@@ -0,0 +1,238 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * See https://spdx.org/licenses/BSD-3-Clause
+ *
+ */
+/* Test Case Description: Tests fam_context_open, fam_context_close and
+ * fam_progress operations for multithreaded model.
+ */
+
+#include <fam/fam.h>
+#include <fam/fam_exception.h>
+
+#include <iostream>
+#include <limits.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "common/fam_test_config.h"
+
+using namespace std;
+using namespace openfam;
+fam *my_fam;
+Fam_Options fam_opts;
+
+Fam_Region_Descriptor *testRegionDesc;
+const char *testRegionStr;
+int rc;
+
+#define NUM_THREADS 32
+#define TOTAL_NUM_THREADS 33
+#define REGION_SIZE (2000 * 1024 * NUM_THREADS)
+#define NUM_CONTEXTS 8
+#define REGION_PERM 0777
+#define NUM_IO_ITERATIONS 50
+fam_context *ctx[NUM_CONTEXTS];
+int activethreads = NUM_THREADS;
+pthread_barrier_t barrier;
+typedef struct {
+    Fam_Descriptor *item;
+    uint64_t offset;
+    int32_t tid;
+    int32_t deltaValue;
+} ValueInfo;
+
+void *thr_check_fam_progress(void *arg) {
+    uint64_t progress = 0;
+    pthread_barrier_wait(&barrier);
+    while (activethreads != 0) {
+        for (int i = 0; i < NUM_CONTEXTS; i++) {
+            try {
+                progress = ctx[i]->fam_progress();
+                cout << "I/Os in Progress for context" << i << " is "
+                     << progress << endl;
+            } catch (Fam_Exception &e) {
+                cout << "Exception caught while invoking fam_progress()"
+                     << endl;
+                cout << "Error msg: " << e.fam_error_msg() << endl;
+                cout << "Error: " << e.fam_error() << endl;
+            }
+        }
+    }
+
+    pthread_exit(NULL);
+}
+
+void *thrd_fam_context_mt_with_fam_progress(void *arg) {
+    ValueInfo *addInfo = (ValueInfo *)arg;
+    Fam_Descriptor *item = addInfo->item;
+    int i = 0;
+    i = addInfo->tid % NUM_CONTEXTS;
+
+    pthread_barrier_wait(&barrier);
+    try {
+        for (int j = 0; j < NUM_IO_ITERATIONS; j++) {
+            int *local1 = (int *)malloc(1000000);
+            int *local2 = (int *)malloc(1000000);
+            ctx[i]->fam_get_nonblocking(local1, item, 500, (i + 1) * 100000);
+            ctx[i]->fam_get_nonblocking(local2, item, 10, (i + 1) * 100000);
+        }
+        ctx[i]->fam_quiet();

```

Comment:
Here, we may end up calling fam_quiet from multiple in same context. Do you want to consider changing this logic ?